src/utils/NetworkUtils.ts

```diff
@@ -89,3 +91,60 @@ export function chainIsOPStack(chainId: number): boolean {
 export function chainIsArbitrum(chainId: number): boolean {
   return [CHAIN_IDs.ARBITRUM, CHAIN_IDs.ARBITRUM_GOERLI, CHAIN_IDs.ARBITRUM_SEPOLIA].includes(chainId);
 }
+
+/**
+ * Determines whether a chainId is associated with another chainId. This is useful for determining if a chainId is associated with a mainnet implementation.
+ * @param mainnetChainId The mainnet chainId to check if the chainId is associated with.
+ * @param chainId The chainId to check if it is associated with the mainnetChainId.
+ * @param stackAssociations If true, will check for associations between stack implementations. Default is false. Currently only used for OP stack implementations.
+ * @returns True if the chainId is associated with the mainnetChainId, otherwise false.
+ * @throws If the mainnetChainId is not a valid mainnetChainId.
+ * @example chainIsAssociatedWith(CHAIN_IDs.MAINNET, CHAIN_IDs.GOERLI) // true
+ * @example chainIsAssociatedWith(CHAIN_IDs.SEPOLIA, CHAIN_IDs.GOERLI) // throws <- Sepolia is not a mainnet implementation
+ * @example chainIsAssociatedWith(CHAIN_IDs.OPTIMISM, CHAIN_IDs.OPTIMISM_GOERLI, true) // true <- Optimism is associated with its testnet implementations
+ * @example chainIsAssociatedWith(CHAIN_IDs.OPTIMISM, CHAIN_IDs.BASE) // false <- Optimism is not associated with Base
+ * @example chainIsAssociatedWith(CHAIN_IDs.OPTIMISM, CHAIN_IDs.BASE, true) // true <- Optimism is associated with Base via the OP stack
+ */
+export function chainIsAssociatedWith(mainnetChainId: number, chainId: number, stackAssociations = false): boolean {
+  if (mainnetChainId === chainId) {
+    // BaseCase: a chain is always associated with itself
+    return true;
+  }
+  // Define a list of loose associations between chainIds. Primarily between main/testnet implementations
+  const associations = {
+    [MAINNET_CHAIN_IDs.OPTIMISM]: [CHAIN_IDs.OPTIMISM_GOERLI, CHAIN_IDs.OPTIMISM_SEPOLIA],
+    [MAINNET_CHAIN_IDs.BASE]: [CHAIN_IDs.BASE_GOERLI, CHAIN_IDs.BASE_SEPOLIA],
+    [MAINNET_CHAIN_IDs.ARBITRUM]: [CHAIN_IDs.ARBITRUM_GOERLI, CHAIN_IDs.ARBITRUM_SEPOLIA],
+    [MAINNET_CHAIN_IDs.ZK_SYNC]: [CHAIN_IDs.ZK_SYNC_GOERLI, CHAIN_IDs.ZK_SYNC_SEPOLIA],
+    [MAINNET_CHAIN_IDs.ARBITRUM]: [CHAIN_IDs.ARBITRUM_GOERLI, CHAIN_IDs.ARBITRUM_SEPOLIA],
+    [MAINNET_CHAIN_IDs.MAINNET]: [CHAIN_IDs.GOERLI, CHAIN_IDs.SEPOLIA],
+    [MAINNET_CHAIN_IDs.POLYGON]: [CHAIN_IDs.MUMBAI, CHAIN_IDs.POLYGON_AMOY],
+    [MAINNET_CHAIN_IDs.LINEA]: [CHAIN_IDs.LINEA_GOERLI],
+    [MAINNET_CHAIN_IDs.BOBA]: [], // Added for completeness. Boba is not associated with any testnet in our system
+  };
```

Comment:
It might make sense to define it outside of the function so it doesn't get instantiated every time it's called.

Reply:
This will tend to grow larger and larger over time though. Do you think it's ultimately sustainable to retain all the mappings in one place? I'm probably OK with the existing `chainIsArbitrum()` and `chainIsOpStack()`.